### DISCUSSION TITLE
Defining labwc behavior when closing a window, possible?

### DISCUSSION THREAD

**Discussion by vanfanel** [labwc/labwc#1553] [Q&A]
Hi there,

Since I am having compositor stalls when closing certain windows (OpenGL programs running on top of MESA Zink with vkBasalt active: I know, crazy thing to do but it works), I was wondering: what does labwc do when closing a window? I mean: what kind of signal does it send? Is it SIGTERM, SIGHUP or SIGKILL?
Is it possible to define that in configuration somehow?

**Comment by ahesford** [comment]
Closing windows sends a close event via the [XDG Shell protocol](https://wayland.app/protocols/xdg-shell#xdg_toplevel:event:close) or, for X applications, a corresponding X11 event. It is up to the application to properly respond to these events.

You can define custom actions for the close button. See the documentation or [how it is configured by default](https://github.com/labwc/labwc/blob/7adf5533f975736be1b1a20b25dcaa650b32c0e0/docs/rc.xml.all#L384).

**Reply by vanfanel** [reply]
Thanks. Isn't it possible to define custom actions for the "close" keyboard shortcut instead?

I have this:

```
<keyboard>
		<keybind key="A-F4"><action name="Close" /></keybind>
</keyboard>
```

Is it possible to define what the "Close" keyboard does under the hood?

**Reply by Consolatis** [reply]
> Is it possible to define what the "Close" keybind does under the hood?

Yes, you could replace it with a `<action name="kill" />` instead. See https://labwc.github.io/labwc-actions.5.html#entry_action_name=kill
However, I am not sure if this helps against the compositor stalls when closing an application, it sounds like your GPU or CPU is busy in this specific condition and thus doesn't let labwc do its job.

**Reply by vanfanel** [reply]
Indeed, using "kill" instead of "close" didn't change anything... Thanks anyway!

**Reply by Consolatis** [reply]
You might be able to run labwc with adjusted niceness (negative) and the game with default or positive niceness. That gives the kernel scheduler a hint of where to spend its spare processing cycles. I don't know if this helps in this case though but it might be worth a try.

For testing you could keep labwc as is and just run the game with `nice -n 10 application`.

**Reply by vanfanel** [reply]
@Consolatis Thanks, tried setting the niceness that way and doesn't make any difference.
Sine it only happens on Intel graphics and not on VC4, I guess this is a driver problem.
